### PR TITLE
feat: support advanced editor in model edit page

### PR DESCRIPTION
**Pull request by HashCookie**
Fix: https://github.com/casbin/casbin-editor/issues/151

### CODE REVIEW

**Comment by HashCookie** [comment]
https://github.com/user-attachments/assets/a417f823-8051-46ff-8d5a-881619bacdd2

File: web/src/ModelEditPage.js [hunk 1]
```diff
@@ -90,6 +86,45 @@ class ModelEditPage extends React.Component {
     });
   }
 
+  submitModelEdit(exitAfterSave) {
```
Comment (hsluoyz) [inline web/src/ModelEditPage.js:89]: Why move position? You lost the git diff

File: web/src/IframeEditor.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React, {forwardRef, useEffect, useImperativeHandle, useRef, useState} from "react";
```
Comment (hsluoyz) [inline web/src/IframeEditor.js:1]: Apache header, note year is 2024

File: web/src/CasbinEditor.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import React, {useCallback, useEffect, useRef, useState} from "react";
```
Comment (hsluoyz) [inline web/src/CasbinEditor.js:1]: Apache header, note year is 2024

File: web/src/index.css [hunk 4]
```diff
@@ -51,3 +51,19 @@ code {
 .custom-link:hover {
   color: rgb(64 64 64) !important;
 }
+
+.full-height-editor {
```
Comment (hsluoyz) [inline web/src/index.css:55]: Don't affect other codemirror usage in Casdoor frontend

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**